### ISSUE TITLE
[Discussion] Problems with our current workflow

### ISSUE THREAD

**Issue by menor** [resmio/git-along#6]
Let's use this issue to comment about things we have encountered during our actual development process, then we can create new issues for each individual problem, if you already have a solution, put it here and we will move it to the newly created issue later.

**Comment by mmarder** [comment]
With bigger projects like the feedback/reputation management its pretty difficult to understand what the current status of the project is. At least for me.

**Comment by menor** [comment]
Agree, I think projects will solve this check [here](https://github.com/resmio/git-along/issues/1#issuecomment-247524027)

**Comment by fabiennepimenta** [comment]
Some of my opinions:

- Gap between design process and development is to big, they should be closer (I'm just talking about the time between my designs and the dev.) –  Projects should solve that like as planing Sprints in advance.

- We keep receiving issues/features from sales & users when we should receive problems  – and of course ideas of how to solve are welcome it if they want – but the core should be the problem and not the solution. 

- Transparency – Sales don't understand what we are doing only with Monday standups. We should start showing some examples of what we are doing and only say the most important things and not a big list to look like we work a lot.  Print some images for eg. and put them in the board – so they can take a look during the brakes.

- To support the previous point, they also don't understand how new features work and their values ( to sell it ) – Demos are a good idea for this. 

- We don't have validations before putting to much effort on issues that we don't know if they will bring value or not to our customers (not for selling).  Personas, Prototypes, MVP and testing are good ideas.

- We don't measure the success of features. Automated newsletters are being used? What about the Tags? Did they stopped to use the 'Notes' field?  Etc... we never know if our deploys have good, bad or any impact on our users. Segment, Analytics, FullStory? Surveys? I don't know..

**Comment by menor** [comment]
All quotes are from @fabiennepimenta 
> Gap between design process and development is to big, they should be closer (I'm just talking about the time between my designs and the dev.) – Projects should solve that like as planing Sprints in advance.

Great point. Let's say I'm a frontend and I see an issue that involves also backend, I should agree with some backend to work on that issue/project and have a quick 10 min chat about how are we going to tackle the issue, and what one person needs from the other. After that both parts of the team are responsible for the task and they should work together on it until completion. If an issue involves UX then also an UX should be present on that quickstart meeting and subsequent update chats. What do you think?

**Comment by menor** [comment]
> We keep receiving issues/features from sales & users when we should receive problems – and of course ideas of how to solve are welcome it if they want – but the core should be the problem and not the solution.
> @fabiennepimenta 
We can have a `feature request` label (or a better name, but I hope you get the idea) to add to that kind of requests we get from customers/other departments. That way we know that the first step when we work with those issues is to research and decide which problem those requests are pointing to, and evaluate possible solutions.

**Comment by menor** [comment]
>Transparency – Sales don't understand what we are doing only with Monday standups. We should start showing some examples of what we are doing and only say the most important things and not a big list to look like we work a lot. Print some images for eg. and put them in the board – so they can take a look during the brakes.
@fabiennepimenta 

Not only sales, I think even us a lot of times have no idea what has been deployed or what hasn't. I'm creating a separate issue for this. #13

**Comment by menor** [comment]
>We don't have validations before putting to much effort on issues that we don't know if they will bring value or not to our customers (not for selling). Personas, Prototypes, MVP and testing are good ideas.
@fabiennepimenta 

I think getting everyone involved at the start of the issue will help with this.

**Comment by menor** [comment]
> We don't measure the success of features. Automated newsletters are being used? What about the Tags? Did they stopped to use the 'Notes' field? Etc... we never know if our deploys have good, bad or any impact on our users. Segment, Analytics, FullStory? Surveys? I don't know..

Opened #14 to discuss this.

**Comment by fabiennepimenta** [comment]
>We keep receiving issues/features from sales & users when we should receive problems – and of course ideas of how to solve are welcome it if they want – but the core should be the problem and not the solution.
@fabiennepimenta

>We can have a feature request label (or a better name, but I hope you get the idea) to add to that kind of requests we get from customers/other departments. That way we know that the first step when we work with those issues is to research and decide which problem those requests are pointing to, and evaluate possible solutions.
@menor 


 - We still have the same problem if they keep asking for features. This is something that they should learn / understand that the most efficient is to know (and actually ask our customers what the problem is) to be able to report it to us instead of solutions. 


––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––


>We don't have validations before putting to much effort on issues that we don't know if they will bring value or not to our customers (not for selling). Personas, Prototypes, MVP and testing are good ideas.
@fabiennepimenta 

>I think getting everyone involved at the start of the issue will help with this.
@menor 


 - Agree, but the important here is also to validate more with our customers.

**Comment by JanNash** [comment]
@fabiennepimenta 
> - We still have the same problem if they keep asking for features. This is something that they 
> should learn / understand that the most efficient is to know (and actually ask our customers what 
> the problem is) to be able to report it to us instead of solutions.

Maybe it'd make sense to make a 30 min / one hour workshop with them to train that mindset. Some times, it's easy to formulate the problem by just asking oneself 'What is the problem' (d'uh :D)

**Comment by menor** [comment]
+1 to the workshop

About the testing, we are doing the internal test group, right? We just need to figure out how to do it with our customers. Do we have any idea on how to assemble the process and structure it? Can you open a new issue for it @fabiennepimenta?

**Comment by ihucos** [comment]
- PRs can take a long time to get merged.
- I'd actually like if we can as responsible persons directly push or merge into master, for example when adding a a new Makefile command or fixing a comment. I often ignore smaller improvements in the code.
- Some issues that e.G. I put some effort in get lost.
- Also already said: As a developer I don't understand why the requirements we should implement matter.

**Comment by menor** [comment]
@ihucos what do you mean in your last point? Sorry, I must still be on holidays :)
